services/static-webserver/client/source/class/osparc/auth/ui/RequestAccount.js

```diff
@@ -240,41 +245,102 @@ qx.Class.define("osparc.auth.ui.RequestAccount", {
       grp.add(buttons)
 
       // interaction
-      submitBtn.addListener("execute", e => {
-        const validForm = this._form.validate();
-        if (validForm) {
-          const formData = {};
-          Object.entries(this._form.getItems()).forEach(([key, field]) => {
-            const val = field.getValue();
-            if (val && (typeof val === "object") && ("classname" in val)) {
-              formData[key] = val.getModel();
-            } else {
-              formData[key] = val;
-            }
-          });
-          this.__submit(formData);
-        }
-      }, this);
-
+      submitBtn.addListener("execute", () => this.__requestPressed(), this);
       cancelBtn.addListener("execute", () => this.fireDataEvent("done", null), this);
 
       this.add(grp);
     },
 
-    __submit: function(formData) {
-      const msg = this.tr("The request is being processed, you will hear from us in the coming hours");
-      osparc.FlashMessenger.getInstance().logAs(msg, "INFO");
-      this.fireDataEvent("done");
+    __requestPressed: function() {
+      const validForm = this._form.validate();
+      if (validForm) {
+        const formData = {};
+        Object.entries(this._form.getItems()).forEach(([key, field]) => {
+          const val = field.getValue();
+          if (val && (typeof val === "object") && ("classname" in val)) {
+            formData[key] = val.getModel();
+          } else {
+            formData[key] = val;
+          }
+        });
+        this.__submit(formData, this.__captchaField.getValue());
+      }
+    },
 
+    __submit: function(formData, captchaValue) {
       const params = {
         data: {
-          "form": formData
+          "form": formData,
+          "captcha": captchaValue
         }
       };
-      osparc.data.Resources.fetch("auth", "postRequestAccount", params);
+      osparc.data.Resources.fetch("auth", "postRequestAccount", params)
+        .then(() => {
+          const msg = this.tr("The request is being processed, you will hear from us in the coming hours");
+          osparc.FlashMessenger.getInstance().logAs(msg, "INFO");
+          this.fireDataEvent("done");
+        })
+        .catch(err => {
+          console.error(err);
+          osparc.FlashMessenger.logAs(err.message, "ERROR");
+          this.__restartCaptcha();
+        });
+    },
+
+    __createCaptchaLayout: function() {
+      const captchaGrid = new qx.ui.layout.Grid(5, 5);
+      captchaGrid.setColumnAlign(0, "center", "bottom");
+      captchaGrid.setColumnFlex(1, 1);
+      const captchaLayout = new qx.ui.container.Composite(captchaGrid);
+
+      const captchaImage = this.__captchaImage = new qx.ui.basic.Image().set({
+        allowShrinkX: true,
+        allowShrinkY: true,
+        scale: true,
+        width: 140,
+        height: 45,
+        cursor: "pointer"
+      });
+      this.__captchaImage.addListener("tap", () => this.__restartCaptcha(), this);
```

Comment:
While this is a nice feature. I don't see any hint for the user to click to regenerate the code. I would add some words with a button to make it more obvious. Usually these are pretty clear

Reply:
Reload button added